### ISSUE TITLE
Difference in implementation of batch normalization between theano and tensorflow

### ISSUE THREAD

**Issue by tearsland123** [tensorflow/tensorflow#22985]
Dear Sir: 
  I have a misunderstanding in the process of batch normalization. e.g.
  The out put of convolution layer 
  [[[102. 138.]
  [246. 282.]]]
  [[[ 507.  624.]
  [ 975. 1092.]]]
  [[[ 912. 1110.]
  [1704. 1902.]]]
  I set the following config of batch norm
  gamma  = [1,2,3]
  beta = [0,0,0]
  mean = [2,3,4]
  Inv = [1/3,1/4,1/5]

  Norm.append(gamma)
  Norm.append(beta)
  Norm.append(mean)
  Norm.append(inv)
 

  model.add(tf.keras.layers.BatchNormalization( momentum=1.0, epsilon=0,center = True,scale = True,trainable=False,renorm_momentum=1.0))
  model.layers[1].set_weights(Norm)
   
   According to the definition of batch_norm , (x - mean/ inv)*gamma + beta, it should generate： 
   
  [[[[   300.    408.]
   [   732.    840.]]

  [[  4032.   4968.]
   [  7776.   8712.]]

  [[ 13620.  16590.]
   [ 25500.  28470.]]]]
  
   However, the final result is
   [[[173.20248 235.55537]
  [422.61404 484.96695]]]
[[[2015.9596 2483.9502]
  [3887.922  4355.9126]]]
[[[ 6090.8965  7419.088 ]
  [11403.661  12731.853 ]]]

  Could anyone tell me what is wrong and how could I get the correct answer?

Best Regards
Allen

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by tearsland123** [comment]
I have not written any custom code
OS: Ubuntu 16.04
TensorFlow installed from pip install 
Version: 1.11.0
Cuda 8.0
cudnn 5.0

**Comment by tearsland123** [comment]
I have not written any custom code
OS: Ubuntu 16.04
TensorFlow installed from pip install
Version: 1.11.0
Cuda 8.0
cudnn 5.0
PC device

Alfred Sorten Wolf <notifications@github.com> 于2018年10月16日周二 上午4:48写道：

**Comment by tearsland123** [comment]
I try to check the detail implemented with the nn.batch_normalization layer, but I am wondering how could I get the value of mean, inv... within the batch_normalization layer

**Comment by harshini-gadige** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @harshini-gadige: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
